lib/src/iterable/iterable_extensions.dart

```diff
@@ -35,4 +37,72 @@ extension IterableExtensions<E> on Iterable<E> {
     }
     yield last;
   }
+
+  /// Zip the elements of the iterable together with those of the
+  /// given iterable to form a list of lists (of length two) of
+  /// elements of the same as those of the inputs..
+  ///
+  /// For example:
+  ///
+  /// [1, 2].zip([5, 6]) // [[1, 5], [2, 6]]
+  Iterable<List<E>> zip(Iterable<E> iterable) =>
+      zipTo(iterable, (left, right) => [left, right]);
+
+  /// Zip the elements of the iterable together with those of the
+  /// given iterable using the provided callback to combine each pair
+  /// of elements.
+  ///
+  /// If one iterable is shorter than the other, the extra elements
+  /// will be combined with `null`. Therefore, the callback
+  /// must be able to handle `null` for either of its parameters,
+  /// though not both.
+  ///
+  /// For example:
+  ///
+  /// ['Kate', 'Pablo'].zipToContainer(
+  ///   [28, 31],
+  ///   (name, age) => {name: age},
+  /// ); // [{'Kate': 28}, {'Pablo': 31}]
+  Iterable<C> zipTo<T, C>(
+    Iterable<T> iterable,
+    C Function(E, T) combine,
+  ) sync* {
+    final left = iterator;
+    final right = iterable.iterator;
+
+    while (true) {
+      final leftAlive = left.moveNext();
+      final rightAlive = right.moveNext();
+      if (!leftAlive && !rightAlive) {
+        break;
+      }
+
+      yield combine(left.current, right.current);
```

Comment:
would it come in handy to write another version of this that would allow `combine` to be async?

Reply:
Interesting idea. When would that come into play? Maybe if the `combine` function had to make network requests?

Reply:
yeah, that's what I was thinking 🤷‍♀

Reply:
whoops, on another laptop haha